### ISSUE TITLE
Soap body matching on http binding type

### ISSUE THREAD

**Issue by aeoncl** [outofcoffee/imposter#326]
### The issue
Hi 😁 ! i'm using imposter to test an external SOAP Service which i was given the WSDL for

I ran into a small issue: i need to use body matching because there is no soap action or anything defined in it
BUT the binding type is defined as HTTP: <soap11:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>

I noticed Imposter was trying to match the bodies on the Soap:Envelope element because of this condition that puts the Soap:Envelope as the root element in the MessageBodyHolder: 
```kotlin
              //SoapPluginImpl.tk
                val bodyHolder: MessageBodyHolder = when (binding.type) {
                    BindingType.SOAP -> {
                        httpExchange.request.body?.let { body -> SoapUtil.parseSoapEnvelope(body) } ?: run {
                            LOGGER.warn("No request body - unable to parse SOAP envelope")
                            httpExchange.response.setStatusCode(400).end()
                            return@handleRoute
                        }
                    }
                    BindingType.HTTP -> {
                        httpExchange.request.body?.let { body -> SoapUtil.parseRawBody(body) } ?: run {
                            LOGGER.warn("No request body - unable to parse request")
                            httpExchange.response.setStatusCode(400).end()
                            return@handleRoute
                        }
                    }
```

If i change the .WSDL's binding type to soap, it works fine, but i don't really want to make any changes to an external .wsdl in order to test it.

### Potential fix
It would be very cool if we were able to override this in the config file: 
```yaml
#config.yaml
plugin: soap
path: /dac
wsdlFile: /apis/dac/dac.wsdl
bindingType: wsdl
```

What is your opinion on the matter ? I can make a PR if you're interested !

Thank you for your time & work 😁

**Comment by outofcoffee** [comment]
Hi @aeoncl - the implementation looks like a misinterpretation of the HTTP binding transport in the SOAP standard. I think the binding type logic should be expecting the SOAP envelope for both bindings. If there's a need to skip the envelope, we should provide an option in the config file to do this.

**Comment by outofcoffee** [comment]
Fixed in [v3.12.1](https://github.com/outofcoffee/imposter/releases).

**Comment by aeoncl** [comment]
Thank you for the quick fix 😁
